### PR TITLE
Connect sequelize to neon

### PR DESCRIPTION
**Pull request by hopeaisulu**

### CODE REVIEW

**Comment by TilegenGena** [comment]
Nice work! You did great job. I do not have question now.

**Review by TilegenGena — APPROVED** [review]

**Review by TilegenGena — APPROVED** [review]

File: server/src/app.module.ts [hunk 1]
```diff
@@ -3,14 +3,41 @@ import { AppController } from './app.controller';
...
+      autoLoadModels: true,
+      models: [User, Task],
+      synchronize: true,
+      dialectOptions: {
+        ssl: {
+          require: true,
+          rejectUnauthorized: false,
+        },
+      },
+    }),
```
Comment (networkflow) [inline server/src/app.module.ts:36]: nit: if you wanted to make the code really clean, I probably wouldn't put a lot of detailed DB setup code in the main AppModule file: I would try to keep AppModule short (just import the things you need and do basic setup), and probably have another file (maybe `db-setup.ts`) that exports the settings object that you pass to `SequelizeModule.forRoot`.

File: server/src/modules/modules.module.ts
Comment (networkflow) [inline server/src/modules/modules.module.ts:1]: What was your plan with this ModulesModule? Curious to know what your plan was, but from what I can tell I don't think I'd recommend it 🙂

File: server/src/modules/task/task.model.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  public static IN_PROGRESS = 'In Progress';
+  public static STUCK = 'Stuck';
+  public static DONE = 'Done';
+
+  public static TASK_STATUS = DataType.ENUM(
+    Task.TO_DO,
+    Task.IN_PROGRESS,
+    Task.STUCK,
+    Task.DONE,
+  );
```
Comment (networkflow) [inline server/src/modules/task/task.model.ts:25]: nit: Small detail, but I personally wouldn't recommend making the `TASK_STATUS` enum strings public static members of the Task class (it adds clutter to the Task class): I would recommend having a TypeScript enum `TaskStatus`, and then feed the values of that enum into `DataType.ENUM` for Sequelize. There's a good SO answer here: https://stackoverflow.com/a/73655097

File: server/src/modules/task/task.model.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  DataType,
+  ForeignKey,
+} from 'sequelize-typescript';
+import { User } from '../user/user.model';
+
+@Table({
+  tableName: 'task',
+  timestamps: true,
+})
+export class Task extends Model<Task> {
```
Comment (networkflow) [inline server/src/modules/task/task.model.ts:14]: I'm not familiar with Sequelize but overall your setup here seems good, nice work!

File: server/src/modules/task/task.service.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    user.firstName = createUserDto.username;
+    user.email = createUserDto.email;
+
+    return user.save();
+  }
+
+  async findAllUsers(): Promise<User[]> {
+    return User.findAll();
+  }
+}
```
Comment (networkflow) [inline server/src/modules/task/task.service.ts:17]: I think this is a mistake: this is all user-related. Did you mean for this to go in `UserService`?

File: server/src/modules/user/user.model.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  @Column
+  firstName: string;
+
+  @Column
+  lastName: string;
+
+  @Column({
+    unique: true,
+  })
```
Comment (networkflow) [inline server/src/modules/user/user.model.ts:24]: Should this have `allowNull: false`?

**Comment by networkflow** [comment]
Overall looks good! Setting up an ORM and new DB models is definitely tricky. I added some comments but most of them are minor suggestions; hope it helps.

**Pull request merged**